starknet_py/utils/data_transformer/_transformation_context.py

```diff
@@ -0,0 +1,59 @@
+from contextlib import contextmanager
+from typing import List, Generator
+
+from starknet_py.utils.data_transformer._calldata_reader import OutOfBoundsError
+from starknet_py.utils.data_transformer.errors import (
+    InvalidTypeException,
+    InvalidValueException,
+)
+
+
+class TransformationContext:
+    """
+    Holds information about context when transforming data.
+    """
+
+    _namespace_stack: List[str]
+
+    @property
+    def current_entity(self):
+        """
+        Name of currently processed entity.
+
+        :return: transformed path.
+        """
+        return ".".join(self._namespace_stack)
+
+    @contextmanager
+    def nest(self, name: str) -> Generator:
+        """
+        Manager used for maintaining information about names of transformed types.
+
+        :param name: name of transformed entity.
+        """
+        self._namespace_stack.append(name)
+        try:
+            yield
+        except OutOfBoundsError as err:
+            # This way we can precisely inform user what's wrong when reading calldata.
```

Comment:
The primary reason for context manager is to know what went wrong and where. In order to do that you need to save info about what's being processed and wrap some exceptions.

Reply:
Okay, so if the primary focus is error handling, I'd:
* Rename the class. `TransformerContext` makes it seems like there's some more important going on than just error handling.
* Rename `nest` to samething along the lines of `error_handling`, so it can be used like `with error_handling(...)`. Current name kind of implies there's something going on in the context manager when it basically yields, and only catches exceptions.

Reply:
Do you have any ideas for a better name? It is just a context for transforming: keeps track of what is being transformed and where did it came from. 
`nest` refers to nested content in parsed structure, `error_handling` won't make sense there. It has to be named with a name that suggest going deeper. Transformers don't care what actually happens in there, they only care that they transform something 1 level deeper.

Reply:
I get your point, however when looking at code in transformers and having multiple calls to `TransformerContext.nest` implies that there's some deeper logic going on there, that influences the flow of the data transformation. 

At least that's how reading the code of transformers felt for me, especially when i started with them an not `TransformationContext`.

Perhaps `nest` makes sense in the context that this handler can be nested, but it makes the code extra confusing when it's already a bit complicated. I'm not saying we have to use exactly `error_handling`, it could be something like `nest_with_error_handling` or whatever, but I'd like the name to imply to the reader of the transformers, that this is mostly a UX detail that can be ignored when initially inspecting the code. `nest` is just too cryptic for me.

Reply:
And for the name, perhaps we could do `TransformerContextErrorHandler` or something along the lines. It's longer but it conveys the intent better.

Reply:
`TransformerContextErrorHandler` is too long and I don't see the benefit.

As for the name of `nest` - would `push_nested_entity` work for you? It explains better what it does.

Reply:
It's as confusing as `nest`. I don't see the problem with longer, more descriptive names: it's an internal module and it increases readability.

Reply:
@war-in @drknzz please provide your opinions. I will do as majority will decide.

Reply:
With the namespaces being a stack, `push_entity` is definitely better than `nest`.
As for the name of the class, I think `TransformationContext` is okay, as long as there is a pretty descriptive docstring.